drivers/usb/udc/udc_bflb_bl61x.c

```diff
@@ -0,0 +1,1480 @@
+
+			if (group_intstatus & USB_CX_COMABT_INT) {
+				udc_submit_event(dev, UDC_EVT_ERROR, -1);
+				LOG_ERR("Control command abort");
+			}
+
+			/* we use the flag to check in and not get double setup,
+			 * better to miss a setup than
+			 * have the memory leak this creates
+			 */
+			if (group_intstatus & USB_CX_SETUP_INT && !priv->setup_received) {
+				priv->ep_in[0] = false;
+				priv->setup_received = true;
+				udc_bflb_bl61x_ctrl_setup_start(dev);
+			} else if (group_intstatus & USB_CX_SETUP_INT) {
+				LOG_ERR("Double Setup");
+			}
+
+			if (group_intstatus & USB_CX_COMFAIL_INT) {
+				udc_submit_event(dev, UDC_EVT_ERROR, -1);
+				LOG_ERR("Control command Fail");
+			}
+
+			/* clear isr */
+			sys_write32(group_intstatus, cfg->base + USB_DEV_ISG0_OFFSET);
+		}
+		if (dev_intstatus & USB_INT_G1) {
+			group_intstatus = sys_read32(cfg->base + USB_DEV_ISG1_OFFSET);
+			group_intstatus &=  ~sys_read32(cfg->base + USB_DEV_MISG1_OFFSET);
+
+			for (uint8_t i = 0; i < cfg->num_of_eps - 1; i++) {
+				if (group_intstatus & (1U << (i * 2))) {
+					udc_bflb_bl61x_ep_dout_start(
+						dev,
+						udc_get_ep_cfg(
+							dev,
+							(USB_EP_DIR_OUT
+							| udc_bflb_bl61x_fifo_get_ep(dev, i))
+							)
+						);
+				}
+			}
+
+			sys_write32(group_intstatus, cfg->base + USB_DEV_ISG1_OFFSET);
+		}
+		if (dev_intstatus & USB_INT_G2) {
+			group_intstatus = sys_read32(cfg->base + USB_DEV_ISG2_OFFSET);
+			group_intstatus &=  ~sys_read32(cfg->base + USB_DEV_MISG2_OFFSET);
+
+			/* suspended */
+			if (group_intstatus & USB_SUSP_INT) {
+				sys_write32(USB_SUSP_INT, cfg->base + USB_DEV_ISG2_OFFSET);
+
+				udc_bflb_bl61x_fifo_reset_ctrl(dev);
+				udc_bflb_bl61x_fifo_reset(dev, 1);
+				udc_bflb_bl61x_fifo_reset(dev, 2);
+				udc_bflb_bl61x_fifo_reset(dev, 3);
+				udc_bflb_bl61x_fifo_reset(dev, 4);
+
+				udc_set_suspended(dev, true);
+				udc_submit_event(dev, UDC_EVT_SUSPEND, 0);
+			}
+
+			/* resumed */
+			if (group_intstatus & USB_RESM_INT) {
+				sys_write32(USB_RESM_INT, cfg->base + USB_DEV_ISG2_OFFSET);
+				udc_set_suspended(dev, false);
+				udc_submit_event(dev, UDC_EVT_RESUME, 0);
+			}
+
+			if (group_intstatus & USBRST_INT) {
+				sys_write32(USBRST_INT, cfg->base + USB_DEV_ISG2_OFFSET);
+
+				udc_bflb_bl61x_fifo_reset_ctrl(dev);
+				udc_bflb_bl61x_fifo_reset(dev, 1);
+				udc_bflb_bl61x_fifo_reset(dev, 2);
+				udc_bflb_bl61x_fifo_reset(dev, 3);
+				udc_bflb_bl61x_fifo_reset(dev, 4);
+
+				tmp = sys_read32(cfg->base + USB_DEV_SMT_OFFSET);
+				tmp &= ~USB_SOFMT_MASK;
+				if (cfg->speed_idx == UDC_BUS_SPEED_HS) {
+					tmp |= USB_BL61X_TIMER_AFTER_RESET_HS;
+				} else {
+					tmp |= USB_BL61X_TIMER_AFTER_RESET_FS;
+				}
+				sys_write32(tmp, cfg->base + USB_DEV_SMT_OFFSET);
+
+				udc_submit_event(dev, UDC_EVT_RESET, 0);
+			}
+
+			if (group_intstatus & USB_ISO_SEQ_ERR_INT) {
+				sys_write32(USB_ISO_SEQ_ERR_INT, cfg->base + USB_DEV_ISG2_OFFSET);
+
+				udc_submit_event(dev, UDC_EVT_ERROR, -1);
```

Comment:
Probably `-EIO` as it tried some I/O and failed.
```suggestion
				udc_submit_event(dev, UDC_EVT_ERROR, -EIO);
```